packages/engine/Source/Scene/GlobeSurfaceTileProvider.js

```diff
@@ -2280,6 +2287,8 @@ function addDrawCommandsForTile(tileProvider, tile, frameState) {
   surfaceShaderSetOptions.clippedByBoundaries = surfaceTile.clippedByBoundaries;
   surfaceShaderSetOptions.hasGeodeticSurfaceNormals = hasGeodeticSurfaceNormals;
   surfaceShaderSetOptions.hasExaggeration = hasExaggeration;
+  surfaceShaderSetOptions.vertexShadowDarkness =
```

Comment:
This will also not be needed since `vertexShadowDarkness` is a `uniform`.

Reply:
@sanjeetsuhag thanks for looking into this. The purpose for making the GlobeSurfaceShaderSet aware of the `vertexShadowDarkness` is for dynamically changing that property value at run time, as it is visible in the last sandcastle example I shared.
If the scene gets pretty dark (that's the cost of having a more realist terrain), the user might want to decrease the darkness for better exploring the terrain.

In the code, the `vertexShadowDarkness` is passed to the `GlobeSurfaceShaderSet`. There it is checked if it has changed and if it has then a redraw is forced. What do you think? Are there other ways to refresh the terrain when the `vertexShadowDarkness` changes?

Reply:
As said before, that's for passing `vertexShadowDarkness` to `GlobeSurfaceShaderSet` for dynamically changing that value at run time